catalog/dags/providers/provider_api_scripts/auckland_museum.py

```diff
@@ -0,0 +1,214 @@
+"""
+Content Provider:       Auckland War Memorial Museum Tāmaki Paenga Hira
+
+ETL Process:            Use the API to identify all CC licensed media.
+
+Output:                 TSV file containing the media and the
+                        respective meta-data.
+
+Notes:                  https://api.aucklandmuseum.com/
+
+Resource:               https://api.aucklandmuseum.com/
+                        https://github.com/AucklandMuseum/API/wiki/Tutorial
+
+Resource | Requests per second | Requests per day
+-- | -- | --
+/search, /id | 10 | 1000
+/id/media | 10 | 1000
+"""
+import logging
+from datetime import datetime, timedelta
+
+from common.constants import IMAGE
+from common.licenses import get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+def convert_date_format(date_obj) -> str:
+    date = str(date_obj)
+    date = date.replace(" ", "T")
+    date = date + "Z"
+    return date
+
+
+logger = logging.getLogger(__name__)
+
+LANDING_URL = (
+    "https://www.aucklandmuseum.com/collections-research/collections/record/am_"
+)
+
+
+class AucklandMuseumDataIngester(ProviderDataIngester):
+    providers = {
+        "image": prov.AUCKLAND_MUSEUM_IMAGE_PROVIDER,
+    }
+    endpoint = "https://api.aucklandmuseum.com/search/collectionsonline/_search"
+    license_url = "https://creativecommons.org/licenses/by/4.0/"
+    total_amount_of_data = 10000
+    DEFAULT_LICENSE_INFO = get_license_info(license_url=license_url)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.delay = 4
+        self.batch_start = 0
+        self.batch_limit = 2000
+        self.headers = {"Content-Type": "application/json"}
+        if self.date:
+            date_from = datetime.strptime(self.date, "%Y-%m-%d").date()
+            self.date_from = str(date_from)
+            self.date_to = str(date_from + timedelta(days=1))
```

Comment:
```suggestion
            date_from = datetime.strptime(self.date, "%Y-%m-%d")
            self.date_from = date_from.isoformat()
            self.date_to = (date_from + timedelta(days=1)).isoformat()
```

Just one last thing! We're sending just the date strings to the API, so for example `2023-10-01` to `2023-10-02` as the range. I tested this locally and it looks like the API interprets that as any records last modified on 2023-10-01 _or_ 2023-10-02, so a 48 hour range.

If we instead send full timestamps `2023-10-01T00:00:00` to `2023-10-02T00:00:00`, it correctly restricts it to the 24-hr window we want :)

Also, now that the DAG is dated we can count on `self.date` to always be passed (the base class handles that for us). So you can remove the conditional and the `convert_date_format` method too 🎉

Reply:
It took awhile to find the correct method and function to convert `self.date`. Thank you!